### PR TITLE
Change design of interaction modal in web UI

### PR DESCRIPTION
**Pull request by Gargron**
Reduce the amount of text on the dialog to the bare minimum. Ask user for username instead of domain, since they might be more familiar with the full thing and less familiar with the word "domain" and which part that is. "Take me home" sounds like giving up rather than proceeding, so now it's just "Go".

grafik

### CODE REVIEW

**Comment by todoneunl** [comment]
Looks good, although I want to suggest keeping the option to use the domain name as well. This is because of backwards compatibility for users who know what a domain is (it's also less typing).

**Review by ClearlyClaire — COMMENTED** [review]
Looks ok overall, though I'm a bit confused by the callbacks and memoization.

It appears to work as expected, though there are two minor UI issues:
- every keystroke makes the “Go” button change for a split second, which is visually distracting
- the copy only talks about handles now, but there is domain autocompletion, which might be confusing

File: app/javascript/mastodon/features/interaction_modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+      <div className='interaction-modal__lead'>
+        <h3>
+          <span className='interaction-modal__icon'>{icon}</span> {title}
+        </h3>
+        <p>
+          <FormattedMessage
+            id='interaction_modal.to_continue_prompt'
+            defaultMessage='To continue, you need to {action} from your account.'
+            values={{ action }}
+          />
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/interaction_modal/index.tsx:567]: This is *probably* fine, but I'd be careful with this. It's usually way better to have full sentences in translatable strings, even if similar strings end up repeating themselves. This gives translators more context and helps with languages which have a different grammatical structure.

**Comment by vmstan** [comment]
> Looks good, although I want to suggest keeping the option to use the domain name as well. This is because of backwards compatibility for users who know what a domain is (it's also less typing).

This, and also would like it if the hint was a more generic domain name.

**Comment by ClearlyClaire** [comment]
> > Looks good, although I want to suggest keeping the option to use the domain name as well. This is because of backwards compatibility for users who know what a domain is (it's also less typing).
> 
> This, and also would like it if the hint was a more generic domain name. `E.g. username@example.com`

It is still possible to use a domain name, it's just not what's suggested as users are more likely to know their full handle than think about the domain part specifically, and the goal of the redesign was to keep the dialog simple.

**Review by ClearlyClaire — APPROVED** [review]

**Review by oneiros — APPROVED** [review]

**Pull request merged**